core/src/main/java/com/google/cloud/sql/core/DefaultAccessTokenSupplier.java

```diff
@@ -129,18 +129,33 @@ public Optional<AccessToken> get() throws IOException {
                   throw new IllegalStateException("Error refreshing credentials " + credentials, e);
                 }
                 if (credentials.getAccessToken() == null
-                    || credentials.getAccessToken().equals("")) {
+                    || "".equals(credentials.getAccessToken().getTokenValue())) {
                   throw new IllegalStateException("Credentials do not have an access token");
                 }
                 if (credentials.getAccessToken().getExpirationTime() != null
                     && credentials.getAccessToken().getExpirationTime().before(new Date())) {
                   throw new IllegalStateException(
                       "Credentials were refreshed but expiration time is in the past");
                 }
+
+                // Now, attempt to downscope the credentials and refresh again
                 GoogleCredentials downscoped = getDownscopedCredentials(credentials);
-                if (downscoped.getAccessToken() == null || downscoped.getAccessToken().equals("")) {
-                  throw new IllegalStateException(
-                      "Donwscoped credentials do not have an access token");
+                if (downscoped.getAccessToken() == null
```

Comment:
This is a duplication of above largely. Could we extract it into a `validateAccessToken` or similar?

Also, there is another `validateAccessToken(accessToken);` below that includes logging in each exception branch. Shall we consolidate all that validation here? It's important that we surface this error in the logs because the refresh logic might be hiding this exception. Ideally, we match the logging we have below and include a request for users to report the error on #1174.